packages/cli/src/lib/transpile/morph/typedefs.ts

```diff
@@ -56,15 +56,11 @@ export function generateTypeDefs(sourceFile: SourceFile, code: string) {
 
   const knownGenerics: Record<string, string | undefined> = {
     MetaFunction: 'T',
+    SerializeFrom: 'T',
   };
 
   typedefsFromImports.forEach((typeElements, moduleSpecifier) => {
     for (const typeElement of typeElements) {
-      // We only use this in root.tsx and it's better to
-      // reuse the existing LoaderReturnData, so skip it.
-      if (typeElement === 'SerializeFrom') continue;
-
-      // Note: SerializeFrom also needs generic if we stop skipping it.
       const hasGeneric = !!knownGenerics[typeElement];
```

Comment:
This ensures that `import type {SerializeFrom} '@shopify/remix-oxygen'` is translated correctly to a JSDoc import with a generic (it doesn't work without a generic):

```js
/**
 * @template T @typedef {import('@shopify/remix-oxygen').SerializeFrom<T>} SerializeFrom
 */
```

We do the same for MetaFunction, which also requires a generic.